WooCommerce/Classes/ViewRelated/Orders/Order Create/OrderFormDataSource.swift

```diff
@@ -0,0 +1,187 @@
+            Section(category: .customerInformation, title: Localization.customerHeader, rows: [.addCustomer]),
+            Section(category: .notes, title: Localization.orderNotesHeader, rows: [.addOrderNote])
+        ]
+    }
+
+    func registerTableViewCells(_ tableView: UITableView) {
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+
+        let headerType = TwoColumnSectionHeaderView.self
+        tableView.register(headerType.loadNib(), forHeaderFooterViewReuseIdentifier: headerType.reuseIdentifier)
+    }
+}
+
+extension OrderFormDataSource: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let section = sections[indexPath.section]
+        let reuseIdentifier = section.rows[indexPath.row].reuseIdentifier
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        configure(cell, section: section, indexPath: indexPath)
+        return cell
+    }
+
+    func heightForHeaderInSection(_ section: Int, tableView: UITableView) -> CGFloat {
+        // Hide header for summary
+        if sections[section].category == .summary {
+            return CGFloat.leastNormalMagnitude
+        }
+
+        return UITableView.automaticDimension
+    }
+
+    func viewForHeaderInSection(_ section: Int, tableView: UITableView) -> UIView? {
+        let reuseIdentifier = TwoColumnSectionHeaderView.reuseIdentifier
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: reuseIdentifier) as? TwoColumnSectionHeaderView else {
+            return nil
+        }
+
+        header.leftText = sections[section].title
+        header.rightText = nil
+
+        return header
+    }
+}
+
+private extension OrderFormDataSource {
+    func configure(_ cell: UITableViewCell, section: Section, indexPath: IndexPath) {
+        let row = section.rows[indexPath.row]
+        switch cell {
+        case let cell as SummaryTableViewCell:
+            configureSummary(cell: cell)
+        case let cell as LeftImageTableViewCell where row == .addOrderItem:
+            configureAddItem(cell: cell)
+        case let cell as LeftImageTableViewCell where row == .addCustomer:
+            configureAddCustomer(cell: cell)
+        case let cell as LeftImageTableViewCell where row == .addOrderNote:
+            configureAddNote(cell: cell)
+        default:
+            fatalError("Unknown cell in Order Form")
+        }
+    }
+
+
+    private func configureSummary(cell: SummaryTableViewCell) {
+        // TODO: add 'new draft' state
+    }
+
+    private func configureAddItem(cell: LeftImageTableViewCell) {
```

Comment:
nit: not needed to declare it as `private`, since the extension is already `private`.